lib/widgets/app.dart

```diff
@@ -143,16 +146,31 @@ class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
           // like [Navigator.push], never mere names as with [Navigator.pushNamed].
           onGenerateRoute: (_) => null,
 
-          onGenerateInitialRoutes: (_) {
+          onGenerateInitialRoutes: (initialRoute) {
+            Route<dynamic>? notificationRoute;
+            int? accountId;
+            if (initialRoute != '/') {
+              final url = Uri.tryParse(initialRoute);
+              if (url case Uri(scheme: 'zulip', host: 'notification')) {
+                final result = NotificationDisplayManager.getRouteForNotification(context, url);
+                if (result != null) {
+                  (notificationRoute, accountId) = result;
+                }
+              }
+            }
             return [
               MaterialWidgetRoute(page: const ChooseAccountPage()),
-              if (initialAccountId != null) ...[
+              if (notificationRoute != null && accountId != null)...[
+                HomePage.buildRoute(accountId: accountId),
+                InboxPage.buildRoute(accountId: accountId),
+                notificationRoute,
+              ] else if (initialAccountId != null) ...[
                 HomePage.buildRoute(accountId: initialAccountId),
                 InboxPage.buildRoute(accountId: initialAccountId),
               ],
```

Comment:
```
notif: Introduce URL based notif open implementation
```

This looks like maybe a behavior change, right? Before, when you opened the app from a notification, the home page and inbox page would be for the first account in the list (the "initial account" in `initialAccountId`), even if that was different from the notification's account. Now, those pages are chosen to match the notification.

I agree the new behavior seems better. Let's mention the change, though, and any other user-visible behavior changes, in the commit message.

Reply:
Removed this behaviour (for now) to keep the diff smaller, this would probably be better done in a separate PR.